go/array_string/min_window/solution.go

```diff
@@ -0,0 +1,101 @@
+// package main has the solution for the minimum window
+// algorithm implementation using the sliding window technique.
+package main
+
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
+func minWindow(mainStr, subStr string) string {
+	// validation
+	if len(mainStr) == 0 || len(subStr) == 0 || len(subStr) > len(mainStr) {
+		return ""
+	}
+	// using the sliding window technique.
+	// create hashmap for required and desired character.
+	required := make(map[string]int)
+	desired := make(map[string]int)
```

Comment:
If you make these `map[rune]int` then you don't have to cast `char` every time

Reply:
Thank you, I understand better.